### ISSUE TITLE
Avoid `ufl_domain` warning

### ISSUE THREAD

**Issue by jwallwork23** [mesh-adaptation/goalie#76]
Running the demos currently raises this warning many times:
```
ufl/ufl/core/expr.py:269: DeprecationWarning: Expr.ufl_domain() is deprecated, please use extract_unique_domain(expr) instead.
```

**Comment by ddundo** [comment]
I don't see this being raised anymore. Could you please double check and close if it's all good?

**Comment by jwallwork23** [comment]
No, I've not noticed it for some time either. Unfortunately, I failed to give a reproducer in the issue, so will just close this. We can reopen it if anything crops up.
